### PR TITLE
feat: [AUTH-1697] (BREAKING) Refactor authenticateJwt and authenticateJwtLocal to use request pattern

### PR DESCRIPTION
**Pull request by max-stytch**
Refactors `authenticateJwt` and `authenticateJwtLocal` manual methods to take in a request object. This is a breaking change, so **DNM** for now. We'll coordinate a breaking change with the dropping of Node 16 later this month.

### CODE REVIEW

File: lib/b2b/sessions.ts [hunk 1]
```diff
@@ -292,6 +292,52 @@ export interface B2BSessionsRevokeResponse {
...
+  /**
+   * If set, return an error if the JWT was issued (based on the "iat" claim) more than max_token_age_seconds seconds ago.
+   * If explicitly set to zero, all tokens will be considered too old, even if they are otherwise valid.
+   */
+  max_token_age_seconds?: number;
+
+  /**
+   * The value used to compare timestamp claims ("exp", "nbf", "iat"). It
+   * defaults to the current date (new Date()).
+   */
```
Comment (jeremy-stytch) [inline lib/b2b/sessions.ts:335]: The pattern in the other comments around here seems to be hard line breaks after full sentences. I kinda like it, so I figure it wouldn't hurt to match.
```suggestion
  /**
   * The value used to compare timestamp claims ("exp", "nbf", "iat").
   * It defaults to the current date (new Date()).
   */
```

File: lib/b2c/sessions.ts [hunk 2]
```diff
@@ -505,6 +505,52 @@ export interface SessionsRevokeResponse {
...
+  /**
+   * If set, return an error if the JWT was issued (based on the "iat" claim) more than max_token_age_seconds seconds ago.
+   * If explicitly set to zero, all tokens will be considered too old, even if they are otherwise valid.
+   */
+  max_token_age_seconds?: number;
+
+  /**
+   * The value used to compare timestamp claims ("exp", "nbf", "iat"). It
+   * defaults to the current date (new Date()).
+   */
```
Comment (jeremy-stytch) [inline lib/b2c/sessions.ts:548]: same as the other file:
```suggestion
  /**
   * The value used to compare timestamp claims ("exp", "nbf", "iat").
   * It defaults to the current date (new Date()).
   */
```

**Review by jeremy-stytch — APPROVED** [review]
Looks good to me for whenever you're ready to coordinate the release

File: lib/b2b/sessions.ts (hunk 1 above)
Comment (max-stytch) [reply lib/b2b/sessions.ts]: Let's do it - https://github.com/stytchauth/stytch-node/pull/253/commits/ebd3cb7eae83ba43668e0e6bae558001fd29a749

**Pull request merged**